pkg/gitprovider/gitlab/gitlab.go

```diff
@@ -260,12 +260,30 @@ func (p *provider) GetCommitURL(repoURL string, sha string) (string, error) {
 }
 
 func convertGitlabMR(glMR gitlab.BasicMergeRequest) gitprovider.PullRequest {
+	var merged = glMR.State == "merged"
+	var mergeCommit string
+	if merged {
+		switch {
+		case glMR.MergeCommitSHA != "":
+			// Regular merge (with or without squash commit),
+			// GitLab gives us the merge commit.
+			mergeCommit = glMR.MergeCommitSHA
+		case glMR.SquashCommitSHA != "":
+			// Fast-forward with squash: no merge commit,
+			// but a squash commit exists.
+			mergeCommit = glMR.SquashCommitSHA
```

Comment:
Since this behavior doesn't exist yet, it would be useful to have the comment reflect that this logic is intended to future-proof incoming changes. Links to relevant issues and discussions for aforementioned incoming changes could also be useful be useful here.